src/pages/images/actions/UploadInstanceBtn.tsx

```diff
@@ -0,0 +1,230 @@
+export interface UploadInstanceFormValues {
+  instanceFile: File | null;
+  name: string;
+  pool: string;
+}
+
+const UploadInstanceBtn: FC = () => {
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+  const { project, isLoading: isProjectLoading } = useProject();
+  const [isUploading, setIsUploading] = useState(false);
+  const [uploadState, setUploadState] = useState<UploadState | null>(null);
+  const eventQueue = useEventQueue();
+  const toastNotify = useToastNotification();
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const [uploadAbort, setUploadAbort] = useState<AbortController | null>(null);
+  const instanceNameAbort = useState<AbortController | null>(null);
+
+  const changeFile = (e: ChangeEvent<HTMLInputElement>) => {
+    const { onChange } = formik.getFieldProps("instanceFile");
+    onChange(e);
+
+    if (e.currentTarget.files) {
+      const file = e.currentTarget.files[0];
+      // remove file extension
+      const instanceName = file.name.split(".tar")[0];
+      const instanceNameSuffix = "-new";
+      void formik.setFieldValue("name", instanceName + instanceNameSuffix);
+      void formik.setFieldValue("instanceFile", file);
+    }
+  };
+
+  const handleCancelUpload = () => {
+    uploadAbort?.abort();
+    closePortal();
+  };
+
+  const handleUpload = async () => {
+    if (!formik.values.instanceFile) {
+      return;
+    }
+
+    setIsUploading(true);
+    const uploadController = new AbortController();
+    setUploadAbort(uploadController);
+
+    try {
+      const instanceName = formik.values.name;
+      const operation = await uploadInstance(
+        formik.values.instanceFile,
+        instanceName,
+        project?.name,
+        formik.values.pool,
+        setUploadState,
+        uploadController,
+      );
+
+      toastNotify.info(
+        <>
+          Backup file upload completed. Now creating the instance{" "}
+          <strong>{instanceName}</strong>.
+        </>,
+      );
+
+      const onSuccess = () => {
+        const message = (
+          <>
+            Created Instance <strong>{instanceName}</strong>.
+          </>
+        );
+
+        const actions = [
+          {
+            label: "Configure",
+            onClick: () =>
+              navigate(`/ui/project/${project?.name}/instance/${instanceName}`),
+          },
+        ];
+
+        toastNotify.success(message, actions);
+      };
+
+      const onFailure = (msg: string) => {
+        toastNotify.failure(
+          "Instance creation from upload failed.",
+          new Error(msg),
+        );
+      };
+
+      const onFinish = () => {
+        void queryClient.invalidateQueries({
+          predicate: (query) => {
+            return query.queryKey[0] === queryKeys.instances;
+          },
+        });
+      };
+
+      eventQueue.set(operation.metadata.id, onSuccess, onFailure, onFinish);
+    } catch (e) {
+      toastNotify.failure("Instance upload failed", e);
```

Comment:
This is causing a generic error:
![image](https://github.com/user-attachments/assets/76f4cc95-0007-4a4b-b9cb-f9ffc3371795)

Where in the error details I found the message "unsupported compression". We should surface that to the user, not just the returned error code.

Reply:
Right, the sync error returned in this case is in `AxiosError` format, which requires a bit of handling to get the message out. Also adjusted this for the axios call in `UploadCustomIso`.